python/src/dataset.rs

```diff
@@ -667,6 +679,42 @@ impl Dataset {
         }
     }
 
+    fn count_unindexed_rows(self_: PyRef<'_, Self>, index_name: String) -> PyResult<Option<usize>> {
+        let idx = RT.block_on(None, self_.ds.load_index_by_name(index_name.as_str()))?;
+        if let Some(index) = idx {
+            RT.block_on(
+                None,
+                self_
+                    .ds
+                    .count_unindexed_rows(index.uuid.to_string().as_str()),
+            )
+            .map_err(|err| PyIOError::new_err(err.to_string()))?
```

Comment:
The error from `block_on` shouldn't require a map_err
```suggestion
?
```